core/src/main/java/com/gentics/mesh/core/endpoint/node/BinaryUploadHandlerImpl.java

```diff
@@ -189,12 +190,21 @@ public void handleUpdateField(InternalActionContext ac, String nodeUuid, String
 			ctx.setHash(hash);
 
 			// Check whether the binary with the given hashsum was already stored
-			HibBinary binary = binaries.findByHash(hash).runInNewTx();
+			Tuple<HibBinary, String> binaryAndUuid = binaries.findByHash(hash)
+					.mapInTx(b -> {
+						return Tuple.tuple(b, b != null ? b.getUuid() : null);
+					})
+					.runInNewTx();
+			HibBinary binary = binaryAndUuid.v1();
+			String binaryUuid = binaryAndUuid.v2();
 
 			// Create a new binary uuid if the data was not already stored
 			if (binary == null) {
 				ctx.setBinaryUuid(UUIDUtil.randomUUID());
 				ctx.setInvokeStore();
+			} else if (binaryUuid != null && fileNotExists(binaryUuid)) {
```

Comment:
Why can't we use UUID from the binary right here without a Tuple?

Reply:
We are not in a transaction context, so binary.getUuid would through a null pointer exception. An alternative would be to open a separate transaction just to get the uuid, but it would bring some overhead

Reply:
Fair.